### PR TITLE
Bookmarking

### PR DESCRIPTION
**Pull request by technige**

### CODE REVIEW

File: neo4j/v1/session.py [hunk 1]
```diff
@@ -336,6 +338,8 @@ def run(self, statement, parameters=None, **kwparameters):
         :return: Cypher result
         :rtype: :class:`.StatementResult`
         """
+        self.last_bookmark = None
```
Comment (lutovich) [inline neo4j/v1/session.py:270]: Do we need to do this initialization together with initialization in line 314?

File: neo4j/v1/session.py [hunk 2]
```diff
@@ -387,15 +391,23 @@ def begin_transaction(self):
...
+            parameters["bookmark"] = bookmark
+
+        self.run("BEGIN", parameters)
         self.transaction = Transaction(self, on_close=clear_transaction)
         return self.transaction
 
     def commit_transaction(self):
-        self.run("COMMIT")
+        result = self.run("COMMIT")
+        self.connection.sync()
```
Comment (lutovich) [inline neo4j/v1/session.py:335]: Is there a way to test that we send commit and rollback messages eagerly?

File: neo4j/v1/bolt.py [hunk 3]
```diff
@@ -238,19 +243,21 @@ def __init__(self, sock, **config):
         # Pick up the server certificate, if any
         self.der_encoded_server_certificate = config.get("der_encoded_server_certificate")
 
+        def on_success(metadata):
+            self.server_version = metadata.get("server")
```
Comment (lutovich) [inline neo4j/v1/bolt.py:255]: `server_version` field is initialized here but where is it read?

File: neo4j/v1/session.py [hunk 4]
```diff
@@ -372,11 +376,11 @@ def close(self):
...
         if self.connection:
             if not self.connection.closed:
-                self.connection.fetch_all()
+                self.connection.sync()
             self.connection.in_use = False
             self.connection = None
 
-    def begin_transaction(self):
+    def begin_transaction(self, bookmark=None):
         """ Create a new :class:`.Transaction` within this session.
```
Comment (lutovich) [inline neo4j/v1/session.py:313]: Should docstring also say something about the bookmark parameter?

**Review by lutovich — COMMENTED** [review]
@nigelsmall review completed, couple small comments.

File: neo4j/v1/session.py (hunk 1 above)
Comment (technige) [reply neo4j/v1/session.py:270]: Ah, yes we do. This clears the bookmark before a run, it's not initialisation.

File: neo4j/v1/session.py (hunk 2 above)
Comment (technige) [reply neo4j/v1/session.py:335]: Tests added for that.

File: neo4j/v1/bolt.py (hunk 3 above)
Comment (technige) [reply neo4j/v1/bolt.py:255]: Just used in a test right now. It will conflict with #108 so I'll adjust it properly for that.

**Pull request merged**